Myfeature

Elzzass opened the pull request:
Pull request

Elzzass committed:
Add second fragment
Add button to general fragment
The button opens the second fragment

Elzzass committed:
Remove second fragment to newScreen

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/GeneralFragment.kt:
```diff
@@ -22,7 +23,16 @@ class GeneralFragment : BaseFragment(), GeneralContractInterface.View {
+        val btn = view.findViewById<View>(R.id.act2_button) as Button
+
+        btn.setOnClickListener {
+            val fragment = SecondFragment()
```
Вынеси это в отдельную функцию в main activity . Потому что этот код может повторяться. И ты в другом место можешь открыть второй фрагмент. Вызывай так (activity as? MainActivity).openCleanScreen

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/GeneralFragment.kt:
```diff
@@ -22,7 +23,16 @@ class GeneralFragment : BaseFragment(), GeneralContractInterface.View {
+
+        btn.setOnClickListener {
+            val fragment = SecondFragment()
+            val fragmentManager = activity!!.supportFragmentManager
```
Использовать !! не рекомендуется. Прилетит KotlinNullPointerException. Приложение крашнится

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/SecondFragment.kt:
```diff
@@ -0,0 +1,30 @@
+import name.bagi.levente.glabrion01.base.view.BaseFragment
+import org.koin.android.ext.android.inject
+
+class SecondFragment : BaseFragment(), GeneralContractInterface.View {
```
Для SecondFragment должен быть свой презентор. Зачем ты используешь тут презентор с другого экрана?

glabrion commented on app/src/main/res/layout/fragment_general.xml:
```diff
@@ -9,4 +9,10 @@
         android:layout_marginTop="@dimen/margin_top"
         android:text="@string/app_name"/>
+    <Button
+        android:id="@+id/act2_button"
```
Давай осмысленное название. act2_button ни о чем не говорит.

glabrion commented on app/src/main/res/layout/fragment_second.xml:
```diff
@@ -0,0 +1,13 @@
+        android:layout_height="wrap_content"
+        android:layout_gravity="center"
+        android:layout_marginTop="@dimen/margin_top"
+        android:text="SECOND FRAGMENT"/>
```
Вынеси текст в ресурсы

glabrion commented on app/src/main/res/layout/fragment_general.xml:
```diff
@@ -9,4 +9,10 @@
+    <Button
+        android:id="@+id/act2_button"
+        android:layout_gravity="bottom|center"
+        android:text="Open second fragment"
```
Вынеси текст в ресурсы. Выдели текст нажми alt+enter и выбери extract to resource . Так кажется

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/GeneralFragment.kt:
```diff
@@ -22,7 +23,16 @@ class GeneralFragment : BaseFragment(), GeneralContractInterface.View {
 
         val view = inflater.inflate(R.layout.fragment_general, container, false)
-
+        val btn = view.findViewById<View>(R.id.act2_button) as Button
```
В котлине можно делать так view?.id_вью

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/SecondFragment.kt:
```diff
@@ -0,0 +1,30 @@
+import name.bagi.levente.glabrion01.R
+import name.bagi.levente.glabrion01.base.view.BaseFragment
+import org.koin.android.ext.android.inject
+
```
И контракт должен быть для этого фрагмента. Грубо говоря в контракте ты прописываешь что может делать вью и презентор.

Elzzass committed:
Remove text "second fragment" to resourse new_fragment

Elzzass committed:
Rename act2_button to open_new_fragment

Elzzass committed:
Change !! to ?. in btn.setOnClickListener that open the new fragement

Elzzass committed:
Move block code setOnClickLister that open new fragment to separate function openCleanScreen in MainActivity

Elzzass committed:
Make  view?.'id_view' to view.findViewById that button open new fragment

Elzzass reviewed (commented):
Сделал изменения

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/GeneralFragment.kt:
```diff
@@ -22,7 +27,8 @@ class GeneralFragment : BaseFragment(), GeneralContractInterface.View {
         val view = inflater.inflate(R.layout.fragment_general, container, false)
-
+        val btn = view?.open_new_fragment as Button
+        (activity as? MainActivity)?.openCleanScreen(btn)
```
не передавай btn в активити. 
сделай лучше так: 
view?.open_new_fragment.setOnClickListener {
(activity as? MainActivity)?.openCleanScreen()
}

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/newScreen/SecondFragment.kt:
```diff
@@ -0,0 +1,32 @@
+        savedInstanceState: Bundle?
+    ): View? {
+
+        generalPresenter.attach(this)
```
generalPresenter относится к General screen. Не надо инжектить его тут

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/newScreen/SecondFragment.kt:
```diff
@@ -0,0 +1,32 @@
+
+class SecondFragment : BaseFragment(),
+    GeneralContractInterface.View {
+    private val generalPresenter: GeneralContractInterface.Presenter by inject()
```
Есть понимание что тут происходит?

glabrion commented on app/src/main/java/name/bagi/levente/glabrion01/main/screen/general/GeneralFragment.kt:
```diff
@@ -22,7 +23,16 @@ class GeneralFragment : BaseFragment(), GeneralContractInterface.View {
+        btn.setOnClickListener {
+            val fragment = SecondFragment()
+            val fragmentManager = activity!!.supportFragmentManager
+            val fragmentTransaction = fragmentManager.beginTransaction()
```
не создавай дополнительно еще переменную fragmentTransaction . можешь сделать так supportFragmentManager.beginTransaction()....

Elzzass committed:
Delete generalPresener inject from second fragment

Elzzass committed:
Change on general fragment
view?.open_new_fragment?.setOnClickListener {
       (activity as? MainActivity)?.openCleanScreen()
       }

Elzzass reviewed (commented):
Ок

Elzzass commented:
Сделал изменения
